lib/gui/app/components/image-selector/image-selector.jsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const React = require('react')
+const ReactDOM = require('react-dom')
+const propTypes = require('prop-types')
+
+const middleEllipsis = require('./../../utils/middle-ellipsis')
+
+const { Provider, Button } = require('rendition')
+
+const shared = require('/./../../../../../lib/shared/units')
+const { StepButton, StepNameButton, StepSelection,
+  Footer, Underline, SizeText, ChangeButton } = require('./../../styled-components')
+
+class SelectImageButton extends React.Component {
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      chosenImage: '',
+      imageSize: ''
+    }
+  }
+
+  static getDerivedStateFromProps (props, state) {
+    state.chosenImage = middleEllipsis(props.getImageName || props.getImageBasename , 20)
+    state.imageSize = shared.bytesToClosestUnit(props.getImageSize)
+    return state
```

Comment:
I think that this method should return a new object and React is going to do the assignments itself. How about something like the following:
```
return {
  chosenImage:...,
  imageSize:...,
};
```

Alternatively, since this doesn't seem to need an internal state, you could change this component to `extend React.PureComponent` and then move the calculation inside the `render()` method and drop the state.
See: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state

Reply:
Actually I believe it should return null in my case. But I moved the calculations to render() method and got rid of the `getDerivedStateFromProps`. Looks like I don't even need to extends `PureComponent` instead of `Component`.